### PR TITLE
feat: Navigation Samples

### PR DESCRIPTION
**Pull request by eriklimakc**

### CODE REVIEW

File: UI/Navigation/src/Navigation/Presentation/PageNavigation/PageNavigation.xaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+				   Margin="12"
+				   HorizontalAlignment="Center"
+				   Source="ms-appx:///Assets/logo.png" />
+
+			<!--  The Navigation.Request attached property is used to navigate to the SamplePage  -->
+			<Button uen:Navigation.Request="Sample" Content="Navigate via Xaml" />
+
+            <!-- New button using Navigation via code behind -->
+            <Button Command="{Binding NavigateCommand}"
+                    Content="Navigate via code behind" />
```
Comment (eriklimakc) [inline UI/Navigation/src/Navigation/Presentation/PageNavigation/PageNavigation.xaml:34]:
```suggestion
			<!-- New button using Navigation via code behind -->
			<Button Command="{Binding NavigateCommand}"
					Content="Navigate via code behind" />
```

File: UI/Navigation/src/Navigation/Presentation/PageNavigation/PageNavigation.xaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+            <!-- Button to show Message Dialog -->
+            <Button Content="Show Simple Message Dialog"
+                    Command="{Binding ShowSimpleDialogCommand}" />
```
Comment (eriklimakc) [inline UI/Navigation/src/Navigation/Presentation/PageNavigation/PageNavigation.xaml:38]: Can we have a page dedicated exclusively to Message Dialog (Xaml and Code-behind) and another one for Dialog (Modal and Flyout)?

File: UI/Navigation/src/Navigation/App.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+	{
+
+		var messageDialog = new MessageDialogViewMap(
+			Title: "This is Uno",
+			Content: "Hello Uno.Extensions",
+			Buttons: new[]
+			{
+				new DialogAction(Label: "Yes"),
+				new DialogAction(Label: "No")
+			}
```
Comment (eriklimakc) [inline UI/Navigation/src/Navigation/App.xaml.cs:55]:
```suggestion
			Buttons:
			[
				new DialogAction(Label: "Yes"),
				new DialogAction(Label: "No")
			]
```

File: UI/Navigation/src/Navigation/Presentation/PageNavigation/PageNavigationViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+	private readonly INavigator _navigator;
+
+	public PageNavigationViewModel(INavigator navigator)
+	{
+		_navigator = navigator;
+		NavigateCommand = new AsyncRelayCommand(NavigateToSamplePage);
+		ShowSimpleDialogCommand = new AsyncRelayCommand(ShowSimpleDialog);
+	}
+
+	public ICommand NavigateCommand { get; }
```
Comment (eriklimakc) [inline UI/Navigation/src/Navigation/Presentation/PageNavigation/PageNavigationViewModel.cs:14]:
```suggestion
	public ICommand NavigateCommand => new AsyncRelayCommand(NavigateToSamplePage);
```

**Review by morning4coffe-dev — COMMENTED** [review]
@vatsashah45, @eriklimakc Great sample! I noticed a small issue: when the NavigationView is in mobile width the menu overlaps with the text:
image


It might be a bit of scope creep, but it would be nice to also showcase the use of `TabBar` in mobile width.

File: UI/Navigation/src/Navigation/App.xaml.cs [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+		);
+
+		routes.Register(
+			new RouteMap("", View: views.FindByViewModel<ShellViewModel>(),
+				Nested:
+				[
+					new ("Main", View: views.FindByViewModel<MainViewModel>(),
+						Nested:
+						[
+                            #region Page Navigation
```
Comment (eriklimakc) [inline UI/Navigation/src/Navigation/App.xaml.cs:118]: Let's keep consistency and use tabs for indentation :)

File: UI/Navigation/src/Navigation/Presentation/BreadcrumbNavigation/BreadcrumbControl.xaml [hunk 6]
```diff
@@ -0,0 +1,46 @@
+﻿<UserControl x:Class="Navigation.Presentation.BreadcrumbControl"
```
Comment (eriklimakc) [inline UI/Navigation/src/Navigation/Presentation/BreadcrumbNavigation/BreadcrumbControl.xaml:1]: Let's keep consistency and use tabs for indentation :)

File: UI/Navigation/src/Navigation/Presentation/FlyoutDrawer/FlyoutDrawerPage.xaml [hunk 7]
```diff
@@ -0,0 +1,35 @@
+﻿<Page x:Class="Navigation.Presentation.FlyoutDrawerPage"
```
Comment (eriklimakc) [inline UI/Navigation/src/Navigation/Presentation/FlyoutDrawer/FlyoutDrawerPage.xaml:1]: Let's keep consistency and use tabs for indentation :)

File: UI/Navigation/src/Navigation/Presentation/TabBarWithDataNavigation/SecondTabBarItemWithDataPage.xaml [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+			<RowDefinition Height="Auto" />
+			<RowDefinition Height="*" />
+		</Grid.RowDefinitions>
+		<TextBlock HorizontalAlignment="Center"
+				   Style="{StaticResource TitleLarge}"
+				   Text="Second Tab" />
+        <TextBlock Text="Data persists between both the tabs."
+                   HorizontalAlignment="Center"
+                   VerticalAlignment="Center"
+                   Margin="40" />
```
Comment (eriklimakc) [inline UI/Navigation/src/Navigation/Presentation/TabBarWithDataNavigation/SecondTabBarItemWithDataPage.xaml:21]:
```suggestion
        <TextBlock Text="Data persists between both the tabs."
                   HorizontalAlignment="Center"
                   VerticalAlignment="Center"
                   Margin="40" />
```
```suggestion
		<TextBlock Margin="40"
				   HorizontalAlignment="Center"
				   VerticalAlignment="Center"
				   Text="Data persists between both the tabs." />
```
Let's keep consistency and use tabs for indentation :)

File: UI/Navigation/src/Navigation/Presentation/TabBarNavigation/TabBarItem3.xaml [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+
+	<Grid>
+        <TextBlock HorizontalAlignment="Center"
+				   VerticalAlignment="Center"
+				   FontSize="24"
+				   Text="Tab Three (User Control)" />
+        <TextBlock HorizontalAlignment="Center"
+                   VerticalAlignment="Bottom"
+                   FontSize="16"
+                   Text="Content for Tab 3 is a user control." />
```
Comment (eriklimakc) [inline UI/Navigation/src/Navigation/Presentation/TabBarNavigation/TabBarItem3.xaml:19]:
```suggestion
		<TextBlock HorizontalAlignment="Center"
				   VerticalAlignment="Bottom"
				   FontSize="16"
				   Text="Content for Tab 3 is a user control." />
```
Let's keep consistency and use tabs for indentation :)

File: UI/Navigation/src/Navigation/Presentation/RequestValue/RequestValueSecondPage.xaml [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+			</ListView>
+
+			<TextBlock Style="{StaticResource TitleLarge}" Text="Or type the Entity name:" />
+
+			<StackPanel Spacing="8">
+				<TextBox Width="400"
+						 PlaceholderText="Entity Name"
+						 Text="{Binding Entity.Name, Mode=TwoWay}" />
+                <TextBlock Text="Click on one of the buttons below to navigate back and see the updated value." />
+                <StackPanel Orientation="Horizontal" Spacing="8">
```
Comment (eriklimakc) [inline UI/Navigation/src/Navigation/Presentation/RequestValue/RequestValueSecondPage.xaml:52]:
```suggestion
				<TextBlock Text="Click on one of the buttons below to navigate back and see the updated value." />
				<StackPanel Orientation="Horizontal" Spacing="8">
```
Let's keep consistency and use tabs for indentation :)

File: UI/Navigation/src/Navigation/Presentation/ModalDialog/ModalDialogSecondPage.xaml [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+				   Source="ms-appx:///Assets/logo.png" />
+
+			<TextBlock HorizontalAlignment="Center"
+					   VerticalAlignment="Center"
+					   FontSize="32"
+					   Text="Dialog Navigation" />
+
+            <TextBlock Text="This is a flyout that takes the full screen." />
+
+        </StackPanel>
```
Comment (eriklimakc) [inline UI/Navigation/src/Navigation/Presentation/ModalDialog/ModalDialogSecondPage.xaml:45]:
```suggestion
			<TextBlock Text="This is a flyout that takes the full screen." />
		</StackPanel>
```
Let's keep consistency and use tabs for indentation :)

File: UI/Navigation/src/Navigation/Presentation/ModalDialog/ModalDialogPage.xaml [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+
+			<!--  Show Content Dialog  -->
+			<Button Command="{Binding ShowContentDialogCommand}" Content="Show Content Dialog" />
+
+			<!--  Show Content Dialog with XAML  -->
+			<Button uen:Navigation.Request="!ModalContentDialog" Content="Show Content Dialog with XAML" />
+
+			<!--  Show Disappearing Flyout  -->
+            <Button Content="Flyout requesting data with cancellation (2s)"
+                    Command="{Binding FlyoutRequestingDataWithCancelCommand}" />
```
Comment (eriklimakc) [inline UI/Navigation/src/Navigation/Presentation/ModalDialog/ModalDialogPage.xaml:43]:
```suggestion
			<Button Command="{Binding FlyoutRequestingDataWithCancelCommand}" Content="Flyout requesting data with cancellation (2s)" />
```
Let's keep consistency and use tabs for indentation :)

File: UI/Navigation/src/Navigation/Presentation/ModalDialog/ComplexFlyoutPageTwo.xaml [hunk 13]
```diff
@@ -0,0 +1,40 @@
+﻿<Page x:Class="Navigation.Presentation.ComplexFlyoutPageTwo"
```
Comment (eriklimakc) [inline UI/Navigation/src/Navigation/Presentation/ModalDialog/ComplexFlyoutPageTwo.xaml:1]: Let's keep consistency and use tabs for indentation :)

File: UI/Navigation/src/Navigation/Presentation/ModalDialog/ComplexFlyoutPageOne.xaml [hunk 14]
```diff
@@ -0,0 +1,54 @@
+﻿<Page x:Class="Navigation.Presentation.ComplexFlyoutPageOne"
```
Comment (eriklimakc) [inline UI/Navigation/src/Navigation/Presentation/ModalDialog/ComplexFlyoutPageOne.xaml:1]: Let's keep consistency and use tabs for indentation :)

File: UI/Navigation/src/Navigation/Presentation/MessageDialog/MessageDialogPage.xaml [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+	<Grid utu:SafeArea.Insets="VisibleBounds">
+		<Grid.RowDefinitions>
+			<RowDefinition Height="Auto" />
+			<RowDefinition />
+		</Grid.RowDefinitions>
+
+		<utu:NavigationBar Content="Message Dialog" />
+
+        <TextBlock x:Name="MessageDialogResultText"
+                   Grid.Row="1" />
```
Comment (eriklimakc) [inline UI/Navigation/src/Navigation/Presentation/MessageDialog/MessageDialogPage.xaml:21]:
```suggestion
		<TextBlock x:Name="MessageDialogResultText" Grid.Row="1" />
```
Let's keep consistency and use tabs for indentation :)

File: UI/Navigation/src/Navigation/Presentation/MessageDialog/MessageDialogPage.xaml [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+			<Button uen:Navigation.Request="MyMessage" Content="Show Message Dialog with XAML" />
+
+			<!--  Button to show Message Dialog with code behind  -->
+			<Button Command="{Binding ShowSimpleDialogCommand}" Content="Show Message Dialog with code behind" />
+
+            <!--  Button to show Message Dialog that disappears aftrer 2 secs  -->
+            <Button Content="MessageDialog Codebehind (cancel after 2s)"
+                    Command="{Binding MessageDialogCodebehindCancelCommand}" />
+
+        </StackPanel>
```
Comment (eriklimakc) [inline UI/Navigation/src/Navigation/Presentation/MessageDialog/MessageDialogPage.xaml:43]:
```suggestion
			<!--  Button to show Message Dialog that disappears aftrer 2 secs  -->
			<Button Command="{Binding MessageDialogCodebehindCancelCommand}" Content="MessageDialog Codebehind (cancel after 2s)" />
		</StackPanel>
```
Let's keep consistency and use tabs for indentation :)

File: UI/Navigation/src/Navigation/Presentation/MainPage.xaml [hunk 17]
```diff
@@ -0,0 +1,92 @@
...
+				</muxc:NavigationViewItem>
+
+                <muxc:NavigationViewItem uen:Region.Name="FlyoutDrawer"
+                                         Content="Flyout Drawer">
+                    <muxc:NavigationViewItem.Icon>
+                        <SymbolIcon Symbol="NewWindow" />
+                    </muxc:NavigationViewItem.Icon>
+                </muxc:NavigationViewItem>
+
+                <muxc:NavigationViewItem uen:Region.Name="./ToFromQuery?QueryUser.Id=8a5c5b2e-ff96-474b-9e4d-65bde598f6bc" Content="FromQuery User 1">
```
Comment (eriklimakc) [inline UI/Navigation/src/Navigation/Presentation/MainPage.xaml:68]: Let's keep consistency and use tabs for indentation :)

File: UI/Navigation/src/Navigation/Presentation/MainPage.xaml [hunk 18]
```diff
@@ -0,0 +1,92 @@
...
+				</muxc:NavigationViewItem>
+
+                <muxc:NavigationViewItem uen:Region.Name="BreadcrumbNavigation"
+                                         Content="Breadcrumb Navigation">
+                    <muxc:NavigationViewItem.Icon>
+                        <SymbolIcon Symbol="NewWindow" />
+                    </muxc:NavigationViewItem.Icon>
+                </muxc:NavigationViewItem>
+
+            </muxc:NavigationView.MenuItems>
```
Comment (eriklimakc) [inline UI/Navigation/src/Navigation/Presentation/MainPage.xaml:87]: Let's keep consistency and use tabs for indentation :)

File: UI/Navigation/src/Navigation/Presentation/FlyoutDrawer/ThirdPage.xaml [hunk 19]
```diff
@@ -0,0 +1,40 @@
+﻿<Page x:Class="Navigation.Presentation.ThirdPage"
```
Comment (eriklimakc) [inline UI/Navigation/src/Navigation/Presentation/FlyoutDrawer/ThirdPage.xaml:1]: Let's keep consistency and use tabs for indentation :)

File: UI/Navigation/src/Navigation/Presentation/FlyoutDrawer/SecondPage.xaml [hunk 20]
```diff
@@ -0,0 +1,40 @@
+﻿<Page x:Class="Navigation.Presentation.SecondPage"
```
Comment (eriklimakc) [inline UI/Navigation/src/Navigation/Presentation/FlyoutDrawer/SecondPage.xaml:1]: Let's keep consistency and use tabs for indentation :)

File: UI/Navigation/src/Navigation/Presentation/FlyoutDrawer/NavFlyout.xaml [hunk 21]
```diff
@@ -0,0 +1,22 @@
+﻿<Flyout x:Class="Navigation.Presentation.NavFlyout"
```
Comment (eriklimakc) [inline UI/Navigation/src/Navigation/Presentation/FlyoutDrawer/NavFlyout.xaml:1]: Let's keep consistency and use tabs for indentation :)

File: UI/Navigation/README.md [hunk 22]
```diff
@@ -0,0 +1,12 @@
+# Uno Platform Navigation Samples
+
+This repository provides simple, to-the-point code samples for the [Uno Platform Navigation](https://platform.uno/docs/articles/external/uno.extensions/doc/Learn/Navigation/NavigationOverview.html).
+
+## Samples
+
+<!-- TODO: add links  -->
```
Comment (eriklimakc) [inline UI/Navigation/README.md:7]: @vatsashah45 We can now continue to do this part

File: UI/Navigation/src/Navigation/Presentation/ModalDialog/ModalDialogSecondPage.xaml [hunk 23]
```diff
@@ -0,0 +1,47 @@
...
+		<Grid.RowDefinitions>
+			<RowDefinition Height="Auto" />
+			<RowDefinition />
+		</Grid.RowDefinitions>
+
+		<utu:NavigationBar Content="Flyout Dialog">
+			<utu:NavigationBar.MainCommand>
+				<AppBarButton>
+					<AppBarButton.Icon>
+						<BitmapIcon UriSource="ms-appx:///Assets/Images/back.png" />
```
Comment (eriklimakc) [inline UI/Navigation/src/Navigation/Presentation/ModalDialog/ModalDialogSecondPage.xaml:22]:
```suggestion
						<BitmapIcon UriSource="ms-appx:///Assets/Images/close.png" />
```
Flyouts should use the close icon instead. This applies to all flyouts.

File: UI/Navigation/src/Navigation/Presentation/ModalDialog/ComplexFlyoutPage.xaml [hunk 24]
```diff
@@ -0,0 +1,18 @@
+﻿<Flyout x:Class="Navigation.Presentation.ComplexFlyoutPage"
```
Comment (eriklimakc) [inline UI/Navigation/src/Navigation/Presentation/ModalDialog/ComplexFlyoutPage.xaml:1]: Can we try to get the Complex Flyout better positioned?

## Windows
image

## Others
image

File: UI/Navigation/src/Navigation/Presentation/ContentControlNavigation/CControlNavigationPage.xaml [hunk 25]
```diff
@@ -0,0 +1,44 @@
...
+			<RowDefinition Height="Auto" />
+			<RowDefinition />
+		</Grid.RowDefinitions>
+		<utu:NavigationBar Padding="60,0,0,0" Content="Content Control Navigation" />
+
+		<Grid Grid.Row="1">
+			<Grid.ColumnDefinitions>
+				<ColumnDefinition Width="*" />
+				<ColumnDefinition Width="*" />
+			</Grid.ColumnDefinitions>
```
Comment (eriklimakc) [inline UI/Navigation/src/Navigation/Presentation/ContentControlNavigation/CControlNavigationPage.xaml:23]: On large screens this looks nice, but small ones it's not so pratical:

image

Maybe we could apply this: https://github.com/unoplatform/uno.extensions/blob/main/testing/TestHarness/TestHarness/Ext/Navigation/Apps/Commerce/CommerceProductsPage.xaml#L29-L36

File: UI/Navigation/src/Navigation/Presentation/PageNavigation/PageNavigation.xaml [hunk 26]
```diff
@@ -0,0 +1,36 @@
...
+	  xmlns:utu="using:Uno.Toolkit.UI"
+	  Background="{ThemeResource ApplicationPageBackgroundThemeBrush}"
+	  mc:Ignorable="d">
+
+	<Grid>
+		<Grid.RowDefinitions>
+			<RowDefinition Height="Auto" />
+			<RowDefinition />
+		</Grid.RowDefinitions>
+		<utu:NavigationBar Padding="60,0,0,0" Content="Page Navigation" />
```
[the remark this answers is not in the document]
Comment (kazo0) [reply UI/Navigation/src/Navigation/Presentation/PageNavigation/PageNavigation.xaml:17]: You can take a look at how Gallery addresses this here. It hides the default NavigationView's toggle button and manages its own: https://github.com/unoplatform/Uno.Gallery/blob/ba5d664e6912cb2c8609225144b4c92492a49694/Uno.Gallery/Views/Shell.xaml#L557

File: UI/Navigation/src/Navigation/Presentation/MainPage.xaml [hunk 27]
```diff
@@ -0,0 +1,119 @@
...
+
+		<muxc:NavigationView x:Name="NavigationViewControl"
+							Grid.Row="1"
+							Grid.ColumnSpan="3"
+							uen:Region.Attached="True"
+							IsBackButtonVisible="Collapsed"
+							IsPaneOpen="True"
+							IsSettingsVisible="False"
+							IsPaneToggleButtonVisible="False">
+
```
Comment (eriklimakc) [inline UI/Navigation/src/Navigation/Presentation/MainPage.xaml:51]:
```suggestion
		<TextBlock Text="Navigation Samples"
				   Grid.Row="0"
				   Grid.Column="1"
				   HorizontalAlignment="Center"
				   VerticalAlignment="Center"
				   FontSize="24"
				   FontWeight="Bold"
				   Canvas.ZIndex="10"
				   Margin="0,0,10,0" />

		<muxc:NavigationView x:Name="NavigationViewControl"
							 Grid.Row="1"
							 Grid.ColumnSpan="3"
							 uen:Region.Attached="True"
							 IsBackButtonVisible="Collapsed"
							 IsPaneOpen="True"
							 IsSettingsVisible="False"
							 IsPaneToggleButtonVisible="False">

```

**Review by kazo0 — APPROVED** [review]

**Comment by kazo0** [comment]
@agneszitte this looks good to me, I don't have access to merge it, would you be able to merge? :)

**Review by carldebilly — APPROVED** [review]

**Pull request merged**